internal/admission/admission_test.go

```diff
@@ -149,3 +149,45 @@ func TestMutate(t *testing.T) {
 	assert.Equal(t, gateway.Spec.Servers[2].Tls.Mode, mutatedGateway.Spec.Servers[2].Tls.Mode)
 	assert.NotEqual(t, gateway.Spec.Servers[2].Tls.CredentialName, mutatedGateway.Spec.Servers[2].Tls.CredentialName)
 }
+
+func TestCanMutateCredentialName(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		description string
+		name        string
+		gatewayName string
+		want        bool
+	}{
+		{
+			description: "Should not mutate a credential name",
+			name:        "ns-gatewayname-ahssbs",
+			gatewayName: "gatewayname",
+			want:        false,
+		},
+		{
+			description: "Should not mutate credential name with gateway-name contains hyphens",
+			name:        "ns-gateway-name-sdfjhdfs",
+			gatewayName: "gateway-name",
+			want:        false,
+		},
+		{
+			description: "Should not mutate credential name with gateway-name any number of hyphens",
+			name:        "ns-gateway-name-long-name-sdfjhdfs",
+			gatewayName: "gateway-name-long-name",
+			want:        false,
+		},
+		{
+			description: "Should mutate a user set credential name",
+			name:        "defaultsecret",
+			want:        true,
+		},
+		{
+			description: "Should mutate a user set credential name using a similar format",
+			name:        "another-format-similar",
```

Comment:
I don't think it effects the outcome of the test but we should set gatewayName here and on L182 for completeness, so that `canMutateCredentialName` does take in a valid gateway name instead of empty string